addons/purchase/migrations/10.0.1.2/pre-migration.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import (division as _py3_division,
+                        print_function as _py3_print,
+                        absolute_import as _py3_abs_import)
+
+from openupgradelib import openupgrade
+
+
+def copy_state_to_purchase_lines(cr):
+    openupgrade.logged_query(cr, '''
+       ALTER TABLE purchase_order_line ADD COLUMN state VARCHAR;
```

Comment:
Please isolate this in a separate instruction. And this is not needed to be logged.